swan-lake/learn-the-language/language-basics.md

```diff
@@ -473,6 +473,102 @@ function nameToString(Name nm) returns string {
+
+The template literal results in compilation of the regular expression at compile-time. On the other hand, the ``fromString`` function results in runtime compilation of the regular expression.
+
+Ballerina regular expressions are based on the ECMAScript 2022 specification. It's important to note that Ballerina's regular expression support is based on a subset of ECMAScript. You can refer to the [Ballerina regular expression specification](https://ballerina.io/spec/lang/master/#section_10.1) to find the supported syntax.
+
+Regular expression templates support interpolation, which allows you to dynamically insert values into a regular expression pattern. This is achieved by placing the interpolation token ``${...}`` inside a ReAtom, which is a sub-syntax defined by the regular expression grammar.
+
+For example:
+```ballerina
+function createPattern(string name) {
+    string:RegExp pattern = re `[a-z]{3}|${name}`;
+}
+```
+In this example, the regular expression pattern includes the ``name`` parameter interpolated into the pattern. This allows for dynamic patterns that can change based on input.
+
+### Unicode property Escape
+
+Unicode general category templates can be used to match characters based on their Unicode properties. These templates can be used with the ``\p{...}`` or ``\P{...}`` escape sequences in a regular expression. Here ``\P`` represents the negation of a unicode template.
+
+Below are some examples of unicode general category templates and their meanings,
+
+- ``\p{L}`` matches any letter 
+- ``\p{Ll}`` matches any lower case letter
+- ``\p{N}`` matches any digit
+- ``\P{P}`` matches any non-punctuation character
+
+Ballerina regex also supports unicode script properties that can be used to match characters based on the language. Inorder to represent a unicode script we have to add ``sc=`` inside the curly braces.
+
+Here are some property meanings,
+
+- ``\p{sc=Arabic}`` matches any character in arabic.
+- ``\p{sc=Latin}`` matches any character in latin.
+
+### Non-capturing groups
+
+Ballerina regexp supports non-capturing groups and flags to control the behavior of regular expression patterns. These allow you to group a set of regular expression constructs without capturing the matched text.
+
+non-capturing groups syntax :=  (?[flags][-][flags]:pattern) 
+
+Ballerina supports four flags that can be used to modify the behavior of regular expression patterns:
+- ``i`` (ignore case) flag: This flag makes the regular expression pattern case-insensitive, so that it matches both uppercase and lowercase letters.
+- ``m`` (multiline) flag: This flag changes the behavior of the ``^`` and ``$`` anchors, so that they match the beginning and end of each line (as opposed to the beginning and end of the entire input string).
+- ``s`` (dot all) flag: This flag changes the behavior of the ``.`` character, so that it matches any character including newline characters.
```

Comment:
```suggestion
- ``s`` (dot all) flag: This flag changes the behavior of the ``.`` character so that it matches any character including new-line characters.
```